pkg/databind/pkg/databind/replacer.go

```diff
@@ -37,38 +39,76 @@ func Replace(vals *Values, template interface{}, options ...ReplaceOption) (tran
 	// if neither discovery nor variables, we just return the template as it
 	if len(vals.discov) == 0 {
 		if len(vals.vars) == 0 {
-			//Historycally used to check for variables in template and return error in that case
-			//Now variables in template are allowed as they can be used in flex i.e and rc is meant
-			//to be executed internal/integrations/v4/integration/definition.go
-			//for onDemand = ignoreConfigPathVar(&foundConfigPath)
-			replaceAllSources(template, []discovery.Discovery{{}}, data.Map{}, rc)
-
+			// TRICKY LOGIC HERE
```

Comment:
suggession: for me, the comments are a bit confusing.
WDYT if instead, we create another method, hasPlaceholders. We could document that function simply with something like: replaceAllSources returns error if template contains a placeholder for which we didn't provided replacing value. (or something like this)

Reply:
I'll review it. These comments are the ones that were before we changed the behaviour so I didn't review them.

Reply:
moved in https://github.com/newrelic/infrastructure-agent/pull/1398/commits/daacc18a22fe08d66249e8648d182ca3b063eea6

let me know if it's what you had in mind :)